e2e/testsuite/tx.go

```diff
@@ -93,9 +91,16 @@ func containsMessage(s string, messages []string) bool {
 	return false
 }
 
-// AssertValidTxResponse verifies that an sdk.TxResponse
+// TODO: Maybe Allow specified error code too, see ensure ics20 transfer fails in base test (and there's one more)
+func (s *E2ETestSuite) AssertTxFailure(resp sdk.TxResponse, expectedMsg string) {
```

Comment:
Commenting on the original api, we could take in a [sdkerror](https://github.com/cosmos/cosmos-sdk/blob/main/errors/errors.go#L90), then we could check the error msg and code/codespace?